### PR TITLE
Release bitmap resources immediately

### PR DESCRIPTION
**Pull request by vandrius**
Release  bitmap memory immediately to prevent out of memory exception

### CODE REVIEW

File: signature-pad/src/main/java/com/github/gcacace/signaturepad/views/SignaturePad.java [hunk 1]
```diff
@@ -606,7 +606,7 @@ private void setIsEmpty(boolean newValue) {
     private void ensureSignatureBitmap() {
         if (mSignatureBitmap == null) {
             mSignatureBitmap = Bitmap.createBitmap(getWidth(), getHeight(),
-                    Bitmap.Config.ARGB_8888);
+                    Bitmap.Config.RGB_565 );
```
Comment (gcacace) [inline signature-pad/src/main/java/com/github/gcacace/signaturepad/views/SignaturePad.java]: Why this? In that case the generated bitmap will not have the alpha channel and we are going to sacrifice the transparency.

File: signature-pad/src/main/java/com/github/gcacace/signaturepad/views/SignaturePad.java [hunk 2]
```diff
@@ -615,6 +615,21 @@ private int convertDpToPx(float dp){
...
+            mSignatureBitmapCanvas = null;
+        }
+
+        if(mSignatureBitmap != null){
+            if(!mSignatureBitmap.isRecycled()) {
+                mSignatureBitmap.recycle();
+            }
+            mSignatureBitmap = null;
+        }
+        System.gc();
```
Comment (gcacace) [inline signature-pad/src/main/java/com/github/gcacace/signaturepad/views/SignaturePad.java]: https://stackoverflow.com/a/2414120/938060

File: SignaturePad-Example/build.gradle [hunk 3]
```diff
@@ -1,12 +1,12 @@
 apply plugin: 'com.android.application'
 
 android {
-    compileSdkVersion Integer.parseInt(project.ANDROID_BUILD_SDK_VERSION)
+    compileSdkVersion 25
```
Comment (gcacace) [inline SignaturePad-Example/build.gradle]: Could you please move these changes to a dedicated CR?

**Comment by vandrius** [comment]
Ignore  Jun 16 commits, it for my personal experiments. But clearBitmap() function change is significant, only this change allowed use this component without OutOfMemmory exception.

**Comment by gnanakeethan** [comment]
Are there any works going this?

**Pull request closed without merging**